include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
My understanding is that we would like to avoid calling `memcpy` with `src` set to `NULL`.

At the moment, there are two cases where that could happen.

Case 1: `n` is zero. We are occasionally calling `memcpy` with `src` set to `NULL` _and_ `n` set to zero, expecting that `memcpy` would be a noop in such case.
Case 2: `n` is non-zero. It is a bug that should end up in segv if we call `memcpy` with `src` set to `NULL` and `n` set to a non-zero value.

I agree that we should avoid calling `memcpy` in case 1, since it's technically a UB. OTOH, I would like to continue calling `memcpy` in case 2 to see a segfault.

Hence the suggestion. Does that make sense?